server/src/main/java/org/opensearch/bootstrap/BootstrapChecks.java

```diff
@@ -751,4 +755,25 @@ public BootstrapCheckResult check(BootstrapContext context) {
             );
         }
     }
+
+    /**
+     * Bootstrap check that if a search node contains multiple data paths
+     */
+    static class MultipleDataPathCheck implements BootstrapCheck {
+
+        @Override
+        public BootstrapCheckResult check(BootstrapContext context) {
+            if (NodeRoleSettings.NODE_ROLES_SETTING.get(context.settings()).contains(DiscoveryNodeRole.SEARCH_ROLE)
+                && Environment.PATH_DATA_SETTING.get(context.settings()).size() > 1) {
+                return BootstrapCheckResult.failure("Having multiple data paths in the search node is not allowed");
```

Comment:
Another nitpick on the wording, but something like "Multiple data paths are not allowed for search nodes" is probably a bit clearer and more direct.